opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -23,11 +23,13 @@
     these attributes can be included in the Resource.*
 
 Resource objects are created with `Resource.create`, which accepts attributes
-(key-values). Resource attributes can also be passed at process invocation in
-the :envvar:`OTEL_RESOURCE_ATTRIBUTES` environment variable. You should
-register your resource with the  `opentelemetry.sdk.trace.TracerProvider` by
-passing them into their constructors. The `Resource` passed to a provider is
-available to the exporter, which can send on this information as it sees fit.
+(key-values). Resources should NOT be created via constructor, and working with
```

Comment:
Should we do more to hide the constructor?

Reply:
Any suggestions on how to do this?

Reply:
We do something like this for the SDK's Span: https://github.com/open-telemetry/opentelemetry-python/blob/18761d78e30a4293abd65a897129b57fbf0c0a15/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L415-L419

Not sure if it 's worth it or not though

Reply:
i'd suggest to create a ticket for it and see if anyone picks it up. For 1.0 doc update should be good enough IMO :)

Unless of course @lzchen wants to do it :)

Reply:
Maybe we should leave it in case someone wants to create a `Resource` that does not take attributes from the `DEFAULT_RESOURCE`?